org.flightgear.FlightGear.json

```diff
@@ -0,0 +1,193 @@
+{
+	"id":"org.flightgear.FlightGear",
+	"runtime":"org.kde.Platform",
+	"runtime-version":"5.10",
+	"sdk":"org.kde.Sdk",
+	"command":"flightgear.sh",
+	"rename-icon":"flightgear",
+	"finish-args":[
+		"--share=ipc",
+		"--socket=x11",
+		"--socket=wayland",
+		"--device=dri",
+		"--socket=pulseaudio",
+		"--share=network",
+		"--filesystem=~/.fgfs/"
+	],
+	"build-options": {
+		"cflags": "-O2",
+		"cxxflags": "-O2"
+	},
+	"cleanup":[
+		"/include",
+		"/lib/pkgconfig",
+		"/lib64/pkgconfig",
+		"/lib/cmake",
+		"/share/doc",
+		"/share/man",
+		"*.a",
+		"*.la",
+		"*.cmake"
+	],
+	"modules":[
+		{
+			"name":"boost",
+			"buildsystem":"simple",
+			"build-commands":[
+				"./bootstrap.sh --prefix=/app",
+				"./b2 headers",
+				"./b2 install"
+			],
+			"sources":[
+				{
+					"sha256":"5721818253e6a0989583192f96782c4a98eb6204965316df9f5ad75819225ca9",
+					"type":"archive",
+					"url":"https://sourceforge.net/projects/boost/files/boost/1.66.0/boost_1_66_0.tar.bz2"
+				}
+			],
+			"cleanup":[
+				"/include"
+			]
+		},
+		{
+			"name":"flightgear-data",
+			"buildsystem":"simple",
+			"sources":[
+				{
+					"type":"archive",
+					"url":"https://sourceforge.net/projects/flightgear/files/release-2017.3/FlightGear-2017.3.1-data.tar.bz2",
+					"sha256":"df08b06e88a29a9f80f29186afd54d278636a663281a1b68e8f484bbb403d898",
+					"dest": "fgdata"
+				}
+			],
+			"build-commands": [
+				"mkdir --parents /app/share/flightgear",
+				"mv fgdata/* /app/share/flightgear/"
+			]
+		},
+		"shared-modules/glu/glu-9.0.0.json",
+		{
+			"name":"mesa",
+			"config-opts":[
+				"--enable-osmesa"
+			],
+			"sources":[
+				{
+					"type":"archive",
+					"url":"https://mesa.freedesktop.org/archive/mesa-17.2.0-rc6.tar.gz",
+					"sha256":"ce25345b7da886cdc730299065e2f27ac285eddc75c89595be16cb8ec8497fcc"
+				}
+			]
+		},
+		{
+			"name": "ffmpeg",
+			"config-opts": [
+				"--disable-everything",
+				"--disable-static",
+				"--enable-shared",
+				"--disable-programs",
+				"--disable-doc",
+				"--disable-avdevice",
+				"--disable-avfilter",
+				"--disable-postproc",
+
+				/* Enough to run the base game, might need extending for mod content */
+				"--enable-decoder=bink,binkaudio_rdft,mp3,pcm_s16le,vorbis",
+				"--enable-demuxer=bink,mp3,ogg,pcm_s16le,wav"
+			],
+			"sources": [
+				{
+				"type": "archive",
+				"url": "http://ffmpeg.org/releases/ffmpeg-2.8.14.tar.xz",
```

Comment:
Very old, does it not support 3.4?

Reply:
It doesn't recognize it. I chose 2.8.z due to this forum entry: http://forum.openscenegraph.org/viewtopic.php?t=16424

Reply:
https://git.archlinux.org/svntogit/community.git/tree/trunk/openscenegraph-ffmpeg3.patch?h=packages/openscenegraph

Reply:
That causes a compilation error:
```
/run/build/openscenegraph/src/osgPlugins/ffmpeg/FFmpegDecoderAudio.cpp:103:36: error: ‘CODEC_ID_NONE’ was not declared in this scope
         if (m_context->codec_id == CODEC_ID_NONE)
```

Reply:
Hmm, fine for now but do ensure that this is at least reported upstream?

Reply:
Even though it's a patched, old version? The next version of FlightGear, released in 24 days, will require at least OSG 3.4

Reply:
Oh alright.